src/ec2/model.ts

```diff
@@ -41,14 +42,24 @@ export class Ec2ConnectionManager {
         return new DefaultIamClient(this.regionCode)
     }
 
-    protected async getAttachedPolicies(instanceId: string): Promise<IAM.attachedPoliciesListType> {
+    protected async getAttachedPolicies(instanceId: string): Promise<IAM.AttachedPolicy[]> {
         const IamRole = await this.ec2Client.getAttachedIamRole(instanceId)
-        if (!IamRole) {
+        if (!IamRole?.Arn) {
             return []
         }
-        const iamResponse = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
-
-        return iamResponse.AttachedPolicies ?? []
+        try {
+            const attachedPolicies = await this.iamClient.listAttachedRolePolicies(IamRole.Arn)
+            return attachedPolicies
+        } catch (e) {
+            if (isAwsError(e) && e.code == 'NoSuchEntity') {
+                const errorMessage = `No policies attached to role: ${IamRole.Arn}.`
```

Comment:
Based on the full message, `NoSuchEntity` does not mean the role has no policies:

    2023-07-18 08:59:15 [ERROR]: aws.ec2.openTerminal: [NoSuchEntity: The role with name AWSCloud9SSMInstanceProfile cannot be found.

```suggestion
                const errorMessage = `Attache role does not exist in IAM: ${IamRole.Arn}.`
```